### PR TITLE
Fixes #31 feature request: Add jwt token based authentication for signup and logout

### LINKED ISSUE #31: [Feature Request]: Add jwt token based authentication for signup and login
### Is your feature request related to a problem? Please describe.

Our apis needs to be authenticated so for that we will use jwt-tokens

### Describe the solution you'd like

Reference: https://www.youtube.com/watch?v=Wq9i77BUBlQ
Reference: https://youtu.be/OmT_uNtT7vY

Implement jwt token based authentication

https://dakotaleemartinez.com/tutorials/devise-jwt-api-only-mode-for-authentication/

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by vanamraghu**
1. Gem file is updated with new gem about devise-jwt
2. devise.rb is updated with configuration related to JWT
3. Migration file 20240827052912_add_jti_to_users.rb is added about jti column to users model
4. schema.rb is updated with new columns
5. user.rb is updated with jwt and also revocation strategies

Resolves #31 

### Description

1. Addition of JWT to the login and providing JWT for the logout

### Type of change

- [ ] Feature

### How to test this PR?

**Signup**
CleanShot 2024-08-27 at 12 22 37

**LogIn**
CleanShot 2024-08-27 at 12 28 01

**Accessing JWT from headers **

CleanShot 2024-08-27 at 12 27 05

**Logout**
Attaching the bearer token to the delete request along with the user data should destroy session and display the log out successfully as response

CleanShot 2024-08-27 at 12 30 06

### CODE REVIEW

File: config/initializers/devise.rb [hunk 1]
```diff
@@ -310,4 +310,17 @@
...
+
+  config.jwt do |jwt|
+    jwt.secret = Rails.application.credentials.fetch(:secret_key_base)
+    jwt.dispatch_requests = [
+      ['POST', %r{^/login$}]
+    ]
+    jwt.revocation_requests = [
+      ['DELETE', %r{^/logout$}]
+    ]
+    jwt.expiration_time = 30.minutes.to_i
```
Comment (rt4914) [inline config/initializers/devise.rb:324]: Our product does not require high security therefore you can change this to 1 month.

**Review by rt4914 — CHANGES_REQUESTED** [review]
Really great work @vanamraghu.

One change suggested.

File: config/initializers/devise.rb (hunk 1 above)
Comment (vanamraghu) [reply config/initializers/devise.rb:324]: Changed to month as per the comments

**Review by rt4914 — APPROVED** [review]
LGTM, Great work @vanamraghu

**Pull request merged**